src/fsharp/ilxgen.fs

```diff
@@ -5662,8 +5665,9 @@ and AllocTopValWithinExpr cenv cgbuf cloc scopeMarks v eenv =
 and EmitSaveStack cenv cgbuf eenv m scopeMarks =
     let savedStack = (cgbuf.GetCurrentStack())
     let savedStackLocals,eenvinner = List.mapFold (fun eenv ty -> AllocLocal cenv cgbuf eenv true (ilxgenGlobalNng.FreshCompilerGeneratedName ("spill",m), ty) scopeMarks) eenv savedStack
-    List.iter (EmitSetLocal cgbuf) savedStackLocals;
-    cgbuf.AssertEmptyStack();
+    for ssl in savedStackLocals do
```

Comment:
Can you change or inline "List.iter" (including in local.fs) to get the same perfomance effect?  That would help all F# code without requiring these changes.